File: app/src/main/java/com/huynd/skyobserver/adapters/GridViewPricePerDayAdapter.java
package com.huynd.skyobserver.adapters;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.huynd.skyobserver.R;
import com.huynd.skyobserver.models.PricePerDay;
import com.huynd.skyobserver.utils.RequestHelper;

import static com.huynd.skyobserver.utils.Constants.CONVENIENCE_FEE_IN_K;

/**
 * Created by HuyND on 8/12/2017.
 */

public class GridViewPricePerDayAdapter extends ArrayAdapter<PricePerDay> {
    private LayoutInflater mInflater;

    public GridViewPricePerDayAdapter(Context context) {
        super(context, R.layout.grid_view_price_per_day_item);
        mInflater = LayoutInflater.from(context);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = convertView;
        TextView textViewDay, textViewPrice;
        ImageView imgvAirline;

        if (view == null) {
            view = mInflater.inflate(R.layout.grid_view_price_per_day_item, parent, false);
            imgvAirline = (ImageView) view.findViewById(R.id.image_view_airline);
            textViewDay = (TextView) view.findViewById(R.id.text_view_day);
            textViewPrice = (TextView) view.findViewById(R.id.text_view_price);
            ViewHolder viewHolder = new ViewHolder(imgvAirline, textViewDay, textViewPrice);
            view.setTag(viewHolder);
        } else {
            ViewHolder viewHolder = (ViewHolder) view.getTag();
            imgvAirline = viewHolder.getImageViewAirline();
            textViewDay = viewHolder.getTextViewDay();
            textViewPrice = viewHolder.getTextViewPrice();
        }

        PricePerDay item = getItem(position);

        if (item != null) {
            textViewDay.setText(String.valueOf(item.getDay()));
            textViewPrice.setText(String.valueOf(item.getPriceTotal() / 1000 + CONVENIENCE_FEE_IN_K));

            String carrier = item.getCarrier();
            if (!TextUtils.isEmpty(carrier)) {
                Glide.with(getContext())
                        .load(RequestHelper.airlinesIconUrlBuilder(carrier))
                        .into(imgvAirline);
            } else {
                imgvAirline.setImageDrawable(null);
            }
        } else {
            textViewDay.setText("");
            textViewPrice.setText("");
        }
        return view;
    }

    public static class ViewHolder {
        private ImageView mImageViewAirline;
        private TextView mTextViewDay;
        private TextView mTextViewPrice;

        public ViewHolder(ImageView imgvAirline, TextView tvDay, TextView tvPrice) {
            mImageViewAirline = imgvAirline;
            mTextViewDay = tvDay;
            mTextViewPrice = tvPrice;
        }

        public TextView getTextViewDay() {
            return mTextViewDay;
        }

        public TextView getTextViewPrice() {
            return mTextViewPrice;
        }

        public ImageView getImageViewAirline() {
            return mImageViewAirline;
        }
    }
}

File: app/src/main/java/com/huynd/skyobserver/adapters/GridViewPricePerDayAdapter.kt
package com.huynd.skyobserver.adapters

import android.content.Context
import android.support.annotation.NonNull
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.huynd.skyobserver.R
import com.huynd.skyobserver.entities.PricePerDay
import com.huynd.skyobserver.utils.Constants.Companion.CONVENIENCE_FEE_IN_K
import com.huynd.skyobserver.utils.RequestHelper
import kotlinx.android.synthetic.main.grid_view_price_per_day_item.view.*

/**
 * Created by HuyND on 8/12/2017.
 */

class GridViewPricePerDayAdapter(@NonNull context: Context) :
        ArrayAdapter<PricePerDay>(context, R.layout.grid_view_price_per_day_item) {
    private var mInflater: LayoutInflater = LayoutInflater.from(context)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val viewHolder: ViewHolder
        val view: View

        if (convertView == null) {
            view = mInflater.inflate(R.layout.grid_view_price_per_day_item, parent, false)
            viewHolder = ViewHolder(view.image_view_airline, view.text_view_day, view.text_view_price)
            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = view.tag as ViewHolder
        }

        viewHolder.run {
            val item = getItem(position)
            if (item != null) {
                tvDay.text = item.day.toString()
                tvPrice.text = "${(item.priceTotal / 1000 + CONVENIENCE_FEE_IN_K)}"

                val carrier = item.carrier
                if (!TextUtils.isEmpty(carrier)) {
                    Glide.with(context)
                            .load(RequestHelper.airlinesIconUrlBuilder(carrier))
                            .into(imgvAirline)
                } else {
                    Glide.with(context).clear(imgvAirline)
                }
            } else {
                Glide.with(context).clear(imgvAirline)
                tvDay.text = ""
                tvPrice.text = ""
            }
        }
        return view
    }

    class ViewHolder(val imgvAirline: ImageView, val tvDay: TextView, val tvPrice: TextView)
}
